cmd/root.go

```diff
@@ -0,0 +1,295 @@
+	rootCmd.PersistentFlags().Int64(settings.InactiveDaysBeforeStopOption, defaultInactiveDaysBeforeStop, "max inactivity period before stopping components in RadixRegistrations")
+	rootCmd.PersistentFlags().String(settings.WhitelistOption, "", "custom whitelist of RadixRegistrations to exclude from cleanup. Appended to default, hardcoded whitelist")
+	logLevel, logErr := os.Getenv("LOG_LEVEL"), error(nil)
+	if logErr != nil {
+		logLevel = defaultLogLevel
+	}
+	switch logLevel {
+	case "DEBUG":
+		log.SetLevel(log.DebugLevel)
+	case "ERROR":
+		log.SetLevel(log.ErrorLevel)
+	default:
+		log.SetLevel(log.InfoLevel)
+	}
+}
+
+func getWhitelist() []string {
+	hardcodedWhitelist := []string{
+		"radix-api",
+		"radix-web-console",
+		"radix-vulnerability-scanner",
+		"radix-github-webhook",
+		"radix-canary-golang",
+		"radix-vulnerability-scanner-api",
+		"radix-servicenow-proxy",
+		"radix-networkpolicy-canary",
+		"radix-cost-allocation-api",
+		"canarycicd-test1",
+		"canarycicd-test2",
+		"canarycicd-test3",
+		"canarycicd-test4",
+	}
+	argWhitelist, _ := rootCmd.Flags().GetString(settings.WhitelistOption)
+	whitelist := append(hardcodedWhitelist, strings.Split(argWhitelist, ",")...)
+	return whitelist
+}
+
+func getKubernetesClient() (kubernetes.Interface, radixclient.Interface) {
+	kubeConfigPath := os.Getenv("HOME") + "/.kube/config"
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+
+	if err != nil {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			log.Fatalf("getClusterConfig InClusterConfig: %v", err)
+		}
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("getClusterConfig k8s client: %v", err)
+	}
+
+	radixClient, err := radixclient.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("getClusterConfig radix client: %v", err)
+	}
+
+	log.Printf("Successfully constructed k8s client to API server %v", config.Host)
+	return client, radixClient
+}
+
+func getKubeUtil() (*kube.Kube, error) {
+	kubeClient, radixClient := getKubernetesClient()
+	kubeutil, err := kube.New(kubeClient, radixClient, nil)
+	if err != nil {
+		return nil, err
+	}
+	return kubeutil, nil
+}
+
+func getTooInactiveRrs(kubeClient *kube.Kube, inactivityLimit time.Duration, action string) ([]v1.RadixRegistration, error) {
+	rrs, err := kubeClient.ListRegistrations()
+	if err != nil {
+		return nil, err
+	}
+	rrsForDeletion := make([]v1.RadixRegistration, 0)
+	for _, rr := range rrs {
+		if isWhitelisted(rr) {
+			log.Debugf("RadixRegistration %s is whitelisted, skipping", rr.Name)
+			continue
+		}
+		ra, err := getRadixApplication(kubeClient, rr.Name)
+		if errors.IsNotFound(err) {
+			log.Debugf("could not find RadixApplication %s, continuing...", rr.Name)
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+		namespaces := getRuntimeNamespaces(ra)
```

Comment:
Some namespaces can be `orphaned`. Suggestion to get namespaces by a label `radix-app` or get all namespaces , put them in a `map`.